.github/workflows/binaries.yml

```diff
@@ -40,3 +40,24 @@ jobs:
           asset_path: ormolu.zip
           asset_name: ormolu-${{ matrix.targetOs }}.zip
           asset_content_type: application/zip
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+    name: Test built binaries
+    runs-on: ${{ matrix.os }}
+    needs: build
+    steps:
+      - name: Download and extract binary
+        run: |
+          curl -sL https://github.com/${{ github.repository }}/releases/download/${{ github.event.release.tag_name }}/ormolu-${{ runner.os }}.zip > ormolu.zip
+          7z e ormolu.zip
+      - name: Basic functionality tests
+        run: |
+          ./ormolu --version
+          echo "data A = A" > test.hs
+          ./ormolu -m check test.hs
```

Comment:
Lurker here, sorry. But I just had to mention that this is a very neat looking test.